qdrant-landing/content/articles/sparse-vectors.md

```diff
@@ -0,0 +1,391 @@
+query_vec, query_tokens = compute_vector(query_text)
+query_vec.shape
+
+query_indices = query_vec.nonzero().numpy().flatten()
+query_values = query_vec.detach().numpy()[indices]
+```
+
+In this example, we use the same model for both document and query. This is not a requirement, but it's a simpler approach.
+
+### 5. Retrieving and Interpreting Results
+
+After setting up the collection and inserting sparse vectors, the next critical step is retrieving and interpreting the results. This process involves executing a search query and then analyzing the returned results.
+
+```python
+# Searching for similar documents
+result = client.search(
+        collection_name=COLLECTION_NAME,
+        query_vector=models.NamedSparseVector(
+            name="text",
+            vector=models.SparseVector(
+                indices=query_indices, 
+                values=query_values,
+            ),
+        ),
+        with_vectors=["text"],
+    )
+
+result
+```
+
+In the above code, we execute a search against our collection using the prepared sparse vector query. The `client.search` method takes the collection name and the query vector as inputs. The query vector is constructed using the `models.NamedSparseVector`, which includes the indices and values derived from the query text. This is a crucial step in efficiently retrieving relevant documents. 
+
+Those familiar with the Qdrant API will notice that the extra care taken to be consistent with the existing named vectors API -- this is to make it easier to use sparse vectors in existing codebases. As always, you're still able to apply payload filters, shard keys, and other advanced features you've come to expect from Qdrant.
+
+```python
+ScoredPoint(id=1, version=0, score=3.4292831420898438, payload={}, vector={'text': SparseVector(indices=[2001, 2002, 2010, 2018, 2032, ...], values=[1.0660614967346191, 1.391068458557129, 0.8903818726539612, 0.2502821087837219, ...])}, shard_key=None)
+```
+
+The result, as shown above, is a `ScoredPoint` object containing the ID of the retrieved document, its version, a similarity score, and the sparse vector. The score is a key element as it quantifies the similarity between the query and the document, based on their respective vectors.
+
+To understand how this scoring works, we use the familiar dot product method:
+
+$$\text{Similarity}(\text{Query}, \text{Document}) = \sum_{i \in I} \text{Query}_i \times \text{Document}_i$$
+
+This formula calculates the similarity score by multiplying corresponding elements of the query and document vectors and summing these products. This method is particularly effective with sparse vectors, where many elements are zero, leading to a computationally efficient process. The higher the score, the greater the similarity between the query and the document, making it a valuable metric for assessing the relevance of the retrieved documents.
+
+
+## Qdrant's Dual-Vector Capability: A Two-Stage Retrieval Powerhouse
```

Comment:
we do not have this feature

Reply:
I clarify that this means a single point has both dense and sparse vector and we can do two stage on the client side